engine/combat/appraisals/valere/moonerang.py

```diff
@@ -50,9 +51,9 @@ def execute_timing_sequence(self: Self) -> None:
                 return
 
         if (
-            # if we jump back after a failure
-            combat_manager.read_back_to_slot() is True
-            # or if we kill the boss/enemy
+            # if you can select a character again
+            combat_manager.selected_character is not PlayerPartyCharacter.NONE
```

Comment:
I suppose this means that it will remain in this state if a failed Moonerang is directly followed by an enemy action? Unless this case is already detected, might it cause issues with blocking, or is that handled in another way?

Reply:
based on the current lifecycle, yes, but blocking execution occurs before this check, so it will route to that in those cases.
The code now checks for playerthis simply skips the consideration execution (selecting a character) using the new code so it can now actually check against this:

https://github.com/shenef/SoS-TAS/pull/183/files#diff-3a646d294a4a4c2ce76af0ff76e35b5f7c9954505927c8e6500d7a6b15748d6aR111-R117